packages/m365_defender/data_stream/event/elasticsearch/ingest_pipeline/pipeline_alert.yml

```
@@ -1,345 +1,448 @@
+                }
+            }
+            if (args.length > 0) {
+                ctx.process.executable = args[0];
+                ctx.process.args = args;
+                ctx.process.args_count = args.length;
+            }
+          }
+        }
+
+  # Threat mapping (Mapped to file unless threat category)
+  - set:
+      field: threat.indicator.registry.key
+      copy_from: m365_defender.event.registry.key
+      ignore_empty_value: true
+      tag: set_threat_indicator_registry_key
+      if: ctx.event?.category != null && ctx.event.category.contains('threat')
+  - append:
+      field: threat.indicator.registry.data.strings
+      value: m365_defender.event.registry.value_data
+      allow_duplicates: false
+      tag: append_threat_indicator_registry_data_strings
+      if: ctx.event?.category != null && ctx.event.category.contains('threat')
+  - set:
+      field: threat.indicator.registry.value
+      copy_from: m365_defender.event.registry.value_name
+      tag: set_threat_indicator_registry_value
+      ignore_empty_value: true
+      if: ctx.event?.category != null && ctx.event.category.contains('threat')
+  - set:
+      field: threat.indicator.file.directory
+      copy_from: m365_defender.event.folder_path
+      ignore_empty_value: true
+      tag: set_threat_indicator_file_directory
+      if: ctx.event?.category != null && ctx.event.category.contains('threat')
+  - set:
+      field: threat.indicator.file.hash.sha1
+      copy_from: m365_defender.event.sha1
+      ignore_empty_value: true
+      tag: set_threat_indicator_file_hash_sha1
+      if: ctx.event?.category != null && ctx.event.category.contains('threat')
+  - set:
+      field: threat.indicator.file.hash.sha256
+      copy_from: m365_defender.event.sha256
+      ignore_empty_value: true
+      tag: set_threat_indicator_file_hash_sha256
+      if: ctx.event?.category != null && ctx.event.category.contains('threat')
+  - set:
+      field: threat.indicator.file.name
+      copy_from: m365_defender.event.file.name
+      ignore_empty_value: true
+      tag: set_threat_indicator_file_name
+      if: ctx.event?.category != null && ctx.event.category.contains('threat')
+  - set:
+      field: threat.indicator.file.size
+      copy_from: m365_defender.event.file.size
+      ignore_empty_value: true
+      tag: set_threat_indicator_file_size
+      if: ctx.event?.category != null && ctx.event.category.contains('threat')
+  - set:
+      field: threat.group.name
+      copy_from: m365_defender.event.threat.family
+      tag: set_threat_group_name
+      ignore_empty_value: true
+      if: ctx.event?.category != null && ctx.event.category.contains('threat')
   - script:
       description: Extract threat subtechnique id & name.
       lang: painless
+      tag: script_extract_subtechnique_id_name
       if: ctx.m365_defender?.event?.attack_techniques != null && ctx.m365_defender.event.attack_techniques != '' && ctx.m365_defender.event.attack_techniques instanceof List
```

```suggestion
      if: ctx.m365_defender?.event?.attack_techniques instanceof List
```
If it is a list it is definitionally not null and not ''.